### PR TITLE
Prettier (commands - only)

### PR DESCRIPTION
**Pull request by chrisnojima**
This PR enables prettier in our workflow.

1. There is a pre-commit hook update
1. There are a couple of new yarn commands
```
yarn run prettier // just the settings we use, no input or writing, you can do 
yarn run prettier -- --write routes.js // etc

yarn run prettier-check // does the prettier validation phase
yarn run prettier-write-all // updates all files (hopefully we don't need to do this w/ the pre-commit in place)
```

If you want to see what the output run looks like theres a temp branch off of this here: https://github.com/keybase/client/pull/6972

I also disabled some eslint style specific rules that conflict with this.

Once this gets approved, if you have a branch you want to merge you can just cherry-pick this pr into it and run prettier on your files (or all files) and it should work fine.
I'll make a pr that just runs the output and merge that into this, then merge it down.

I run prettier automatically on save in my vim setup, there's instructions here: https://github.com/prettier/prettier#editor-integration

### CODE REVIEW

**Comment by chrisnojima** [comment]
I was fixing up a bunch of crap in the 'applied' pr but instead i'm going to fix it here so that pr is clean. more commits coming soon

- [ ] lint passes after apply
- [ ] flow passes after apply

**Comment by chrisnojima** [comment]
ok, this PR, if prettier is applied to it, passes all tests so this is all the code changes required in this PR.

@keybase/react-hackers this is ready

File: shared/desktop/renderer/dumb.js [hunk 1]
```diff
@@ -13,15 +13,22 @@ module.hot && module.hot.accept('../../dev/dumb-sheet/render.desktop', render)
 class Wrapper extends React.Component<any, any, any> {
   constructor () {
     super()
-    this.state = {dumbFilter: localStorage['dumbFilter'] || ''} // eslint-disable-line
+    this.state = {
+      // eslint-disable-next-line
+      dumbFilter: localStorage['dumbFilter'] || '',
```
Comment (chrisnojima) [inline shared/desktop/renderer/dumb.js]: doesnt' know about localStorage at all, we could make it a global on top

File: shared/actions/chat/index.js [hunk 2]
```diff
@@ -392,8 +393,9 @@ function _unboxedToMessage (message: ChatTypes.MessageUnboxed, yourName, yourDev
     const payload: ChatTypes.OutboxRecord = message.outbox
     const messageState: Constants.MessageState = (payload && payload.state && payload.state.state === ChatTypes.LocalOutboxStateType.error) ? 'failed' : 'pending'
     const messageBody: ChatTypes.MessageBody = payload.Msg.messageBody
-    // $FlowIssue
-    const failureDescription = messageState === 'failed' ? _decodeFailureDescription(payload.state.error.typ) : null
+    const failureDescription = messageState === 'failed' // prettier-ignore $FlowIssue
+      ? _decodeFailureDescription(payload.state.error.typ)
```
Comment (chrisnojima) [inline shared/actions/chat/index.js]: you'd also havee to check messageState === and i'm worried it'd still wrap lines

File: protocol/diff_test.sh [hunk 3]
```diff
@@ -15,12 +15,12 @@ make
...
 #   protocol/
 #   go/
-#   shared/
+# We ignore shared as we run prettier and it's complicated to sync that up
 # This build process is idempotent. We expect there to be no changes after
 # re-running the protocol generation, because any changes should have been
 # checked in.
-if ! git diff --quiet --exit-code HEAD -- ./ ../go/ ../shared/; then
-  git diff HEAD -- ./ ../go/ ../shared/;
+if ! git diff --quiet --exit-code HEAD -- ./ ../go/; then
```
Comment (chrisnojima) [inline protocol/diff_test.sh]: ticketed: https://keybase.atlassian.net/browse/DESKTOP-3816

**Comment by keybase-ci-visdiff** [comment]
The commits f9b194de6a6d...66494b902b86 introduce visual changes on linux. <details><summary>:mag_right: 8 changed</summary>

 * Changed: **Standard_Screen-Success_w_Custom_Notification_Element** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494943425896-f9b194de6a6d-66494b902b86-linux/Standard_Screen-Success_w_Custom_Notification_Element-f9b194de6a6d.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494943425896-f9b194de6a6d-66494b902b86-linux/Standard_Screen-Success_w_Custom_Notification_Element-66494b902b86.png)  
   Standard_Screen-Success_w_Custom_Notification_Element comparison
 * Changed: **New_Proof_Post-Twitter_Waiting** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494943425896-f9b194de6a6d-66494b902b86-linux/New_Proof_Post-Twitter_Waiting-f9b194de6a6d.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494943425896-f9b194de6a6d-66494b902b86-linux/New_Proof_Post-Twitter_Waiting-66494b902b86.png)  
   New_Proof_Post-Twitter_Waiting comparison
 * Changed: **New_Proof_Post-Twitter_Error** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494943425896-f9b194de6a6d-66494b902b86-linux/New_Proof_Post-Twitter_Error-f9b194de6a6d.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494943425896-f9b194de6a6d-66494b902b86-linux/New_Proof_Post-Twitter_Error-66494b902b86.png)  
   New_Proof_Post-Twitter_Error comparison
 * Changed: **New_Proof_Post-Twitter** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494943425896-f9b194de6a6d-66494b902b86-linux/New_Proof_Post-Twitter-f9b194de6a6d.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494943425896-f9b194de6a6d-66494b902b86-linux/New_Proof_Post-Twitter-66494b902b86.png)  
   New_Proof_Post-Twitter comparison
 * Changed: **New_Proof_Post-HTTP** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494943425896-f9b194de6a6d-66494b902b86-linux/New_Proof_Post-HTTP-f9b194de6a6d.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494943425896-f9b194de6a6d-66494b902b86-linux/New_Proof_Post-HTTP-66494b902b86.png)  
   New_Proof_Post-HTTP comparison
 * Changed: **New_Proof_Post-Facebook** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494943425896-f9b194de6a6d-66494b902b86-linux/New_Proof_Post-Facebook-f9b194de6a6d.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494943425896-f9b194de6a6d-66494b902b86-linux/New_Proof_Post-Facebook-66494b902b86.png)  
   New_Proof_Post-Facebook comparison
 * Changed: **New_Proof_Post-DNS** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494943425896-f9b194de6a6d-66494b902b86-linux/New_Proof_Post-DNS-f9b194de6a6d.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494943425896-f9b194de6a6d-66494b902b86-linux/New_Proof_Post-DNS-66494b902b86.png)  
   New_Proof_Post-DNS comparison
 * Changed: **New_Proof_Post-Coinbase** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494943425896-f9b194de6a6d-66494b902b86-linux/New_Proof_Post-Coinbase-f9b194de6a6d.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494943425896-f9b194de6a6d-66494b902b86-linux/New_Proof_Post-Coinbase-66494b902b86.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1494943425896-f9b194de6a6d-66494b902b86-linux/New_Proof_Post-Coinbase.png)
</summary>

**Pull request merged**